API/index.js

```diff
@@ -200,7 +200,7 @@ class API {
           headers['X-RateLimit-Type'] === THROTTLE_CONSTANTS.RATE_LIMIT_HEADER
         ) {
           // Increment daily tokens on failed request
-          throttle.incrementDailyTokens();
+          await throttle.incrementDailyTokens();
```

Comment:
I noticed that incrementDailyTokens() is allowed even if emptyTokenReservoir has been called. Might this be an issue?

Reply:
Yes, under some circumstances. I'll add a hitDailyLimit flag to the throttle object